Added a way to export our database to a .csv file

solanoalex opened the pull request:
## Summary
This PR adds an export feature to our app.

## Changes Made
- In views.py, created three helper functions that each write one of our models into a .csv file
- In views.py, created an export()  that calls these three helper functions
- In views.py, created a generate_fake_data() to test the export()
- In views.py, created three helper functions that read in .csv file to populate each one of our models
- In views.py, created a clear database function
- In views.py, created a populate() that calls these three helper functions and the clear database function
- In views.py, added the export and import functionality to the footer in the browse page

## Why this Change
This change will allow us to standardize our database and share our database to each other.

## Checklist

- [x] Created three helper functions for export()
- [x] Created the export()
- [x] Created a function to generate fake data 
- [x] Created a function to clear the database before importing
- [x] Created three helper functions for populate()
- [x] Created the populate()
- [x] Added the export and import functionalty to the footer in the browse page

jctjad commented:
Looks great Alex. Clean code 🔥 💯 🔥 

(1) Do you think you could add a function populate() that will populate the .csv files data back into the database?
(2) Can you add a "data" folder to static and store the .csv files in there instead of in the main project root?
(3) Probably best to remove .csv from the .gitignore file if we want to share the .csv files for testing :)
(4) Atm to download the .csv files, we use the /export route. Any chance you could change that to a button (in the footer) instead of a route? Might make the process easier to work with.

Looks great overall though! You did the hard part really well

solanoalex commented:
Finished implementing the four suggestions! In order to resolve the merge conflict, I think accepting the current changes should fix it. Let me know if there other issues!

jctjad commented:
Love the changes! 

Small bug: in your get_User_Data(), get_Item_Data(), get_Chat_Data() functions you search for the .csv file "with open('../marketplace/data/Items.csv', 'w', newline='') as csvfile:", but if the user (me for instance) doesn't already have those .csv files created, the site crashes.

(1) Add a check to see if the files already exist. If they do, your code works, but if they don't, we need to create them.
(2) Check to see if the files have multiple lines before skipping the first line. If we skip a one-line file, site crashes.
(3) Move the data folder inside the static folder!
(4) Maybe generate_Fake_Data() should be an option in the import() function, not export()? Your call but feels more intuitive to me that way.

solanoalex commented:
Alrighty, I added the file existing check, the header existing check, moved the data folder, and added the generate_Fake_Data() as an option in import().
